core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1876,6 +1877,41 @@ public FileInfo createFile(AlluxioURI path, CreateFileContext context)
     }
   }
 
+  /**
+   * @param rpcContext the rpc context
+   * @param inodePath the path to be created
+   * @param context the method context
+   */
+  private void overwriteFileInternal(RpcContext rpcContext, LockedInodePath inodePath,
+                                     CreateFileContext context)
+      throws FileDoesNotExistException, IOException, InvalidPathException,
+      FileAlreadyExistsException {
+    if (inodePath.fullPathExists()) {
+      Inode currentInode = inodePath.getInode();
+      if (!context.getOptions().getOverwrite()) {
```

Comment:
add a check in case the client is in old version
```suggestion
      if (!context.getOptions().hasOverwrite() || !context.getOptions().getOverwrite()) {
```

Reply:
done